crates/uv/src/commands/pip_check.rs

```diff
@@ -42,18 +46,50 @@ pub(crate) fn pip_check(
 
     // Build the installed index.
     let site_packages = SitePackages::from_executable(&venv)?;
+    let packages: Vec<&InstalledDist> = site_packages.iter().collect();
 
-    let mut is_compatible = true;
-    // This loop is entered if and only if there is at least one conflict.
-    for diagnostic in site_packages.diagnostics()? {
-        is_compatible = false;
-        writeln!(printer.stdout(), "{}", diagnostic.message())?;
-    }
+    let s = if packages.len() == 1 { "" } else { "s" };
+    writeln!(
+        printer.stderr(),
+        "{}",
+        format!(
+            "Checked {} in {}",
+            format!("{} package{}", packages.len(), s).bold(),
+            elapsed(start.elapsed())
+        )
+        .dimmed()
+    )?;
 
-    if !is_compatible {
-        return Ok(ExitStatus::Failure);
-    }
+    let diagnostics: Vec<Diagnostic> = site_packages.diagnostics()?.into_iter().collect();
 
-    writeln!(printer.stdout(), "Installed packages pass the check.").unwrap();
-    Ok(ExitStatus::Success)
+    if diagnostics.is_empty() {
+        writeln!(
+            printer.stderr(),
+            "{}",
+            "All installed packages are compatible".to_string().dimmed()
+        )?;
+
+        Ok(ExitStatus::Success)
+    } else {
+        let incompats = if diagnostics.len() == 1 {
+            "incompatibility"
+        } else {
+            "incompatibilities"
```

Comment:
Diagnostics? I don't mind this language, it's just what we use internally.

Reply:
Diagnostic feels too agnostic whereas these are "problems", idk.